### ISSUE TITLE
CI: convert all builds to new Helix machines

### ISSUE THREAD

**Issue by BruceForstall** [dotnet/coreclr#19037]
After https://github.com/dotnet/coreclr/issues/18048, we will use new Helix-provided machines for building ARM32/ARM64.

Convert the rest of the Jenkins CI build pool to use the same machines, so all builds use the same toolset.

Currently, they use the "latest-or-auto" tag (see netci.groovy, `setMachineAffinity()`).

**Comment by BruceForstall** [comment]
@jashook @mmitche @RussKeldorph Do you agree this is the right thing to do?

**Comment by jashook** [comment]
Sounds good to me.

**Comment by BruceForstall** [comment]
Also, @MattGal 

I presume these Helix VMs, used for Windows x86/x64/arm32/arm64 builds, can also be used for Windows x86/x64 test runs.

**Comment by MattGal** [comment]
Absolutely, but if you see slow performance it can be from varying scaling rules for sets that only run builds versus mixed build / runs; ping me if you see that as it's a cheap fix.

**Comment by BruceForstall** [comment]
@MattGal @mmitche It looks like the Helix VMs are quite a bit slower than the machines we're using right now in Jenkins, such that I'm seeing lots of job timeout failures. For one test I checked that completed, the runtime with the Helix VMs is 1.6x the current VMs.

https://github.com/dotnet/coreclr/pull/19388 is testing this.

Looks like there may be other issues as well.

**Comment by mmitche** [comment]
@BruceForstall The issue is that the SKU the Helix VMs doesn't have the same disk I/IO perf as the Jenkins VMs. The Jenkins VMs are an older SKU.  Upgrading to a newer SKU that has equivalent disk perf is significantly more expensive.

We are investigating other sizes that may help (without being too expensive)  Honestly this needs to be fixed by a combo of right hardware and improvements in repo build efficiency.

**Comment by BruceForstall** [comment]
Ok, I'm going to abandon this effort for now.
